src/Analyzers/CSharp/Tests/NamingStyles/NamingStylesTests.cs

```diff
@@ -1293,5 +1293,24 @@ public async Task TestRefactorNotifyTypeParameterNamesStartWithT()
             Assert.Equal("arg", symbolRenamedOperation._symbol.Name);
             Assert.Equal("TArg", symbolRenamedOperation._newName);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
+        [WorkItem(47508, "https://github.com/dotnet/roslyn/issues/47508")]
+        public async Task TestRecordParameter_NoDiagnosticWhenCorrect()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"record Foo(int [|MyInt|]);",
+                new TestParameters(options: s_options.MergeStyles(s_options.PropertyNamesArePascalCase, s_options.ParameterNamesAreCamelCaseWithPUnderscorePrefix)));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
+        [WorkItem(47508, "https://github.com/dotnet/roslyn/issues/47508")]
+        public async Task TestRecordParameter_ParameterFormattedAsProperties()
+        {
+            await TestInRegularAndScriptAsync(
+@"public record Foo(int [|myInt|]);",
+@"public record Foo(int [|MyInt|]);",
+                options: s_options.MergeStyles(s_options.PropertyNamesArePascalCase, s_options.ParameterNamesAreCamelCaseWithPUnderscorePrefix));
+        }
```

Comment:
❕ Need tests for cases where parameters are given using an explicit constructor inside a record.